specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-06-01/Marketplace.json

```diff
@@ -482,6 +482,96 @@
         }
       }
     },
+    "/providers/Microsoft.Marketplace/privateStores/{privateStoreId}/queryApprovedPlans": {
+      "parameters": [
+        {
+          "$ref": "#/parameters/PrivateStoreIdParameter"
+        },
+        {
+          "$ref": "#/parameters/ApiVersionParameter"
+        }
+      ],
+      "post": {
+        "tags": [
+          "PrivateStores"
+        ],
+        "operationId": "PrivateStore_QueryApprovedPlans",
+        "description": "Get map of plans and related approved subscriptions.",
+        "parameters": [
+          {
+            "in": "body",
+            "name": "payload",
+            "schema": {
+              "$ref": "#/definitions/QueryApprovedPlansPayload"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Map of plan ids and their related approved subscriptions.",
+            "schema": {
+              "$ref": "#/definitions/QueryApprovedPlansResponse"
+            }
+          },
+          "default": {
+            "description": "Microsoft.Marketplace error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "QueryApprovedPlans": {
+            "$ref": "./examples/QueryApprovedPlans.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Marketplace/privateStores/{privateStoreId}/bulkCollectionsAction": {
```

Comment:
Why does this need a bulk operation instead of performing the operation on each collection individually?  Bulk operations are problematic because they may perform actions the user did not have RBAC rights to perform (since the operation is not performed directly on the scope it affects) and they don't expose in the activity log everything they are modifying.  It also affects downstream notifications of changes, etc...

Reply:
Currently, we don't have RBAC per collection. It is on post Ni roadmap.
Once we get there, we have 2 options ( on the RP side): 
1. Check RBAC  for each collection and fail the action if there is a least one collection that is not authorized.
2. Check RBAC  for each collection and activate only for the allowed ones. In this case, the response will contain the details for the ones that passed and the ones that failed. The current response already supports this scenario.

Reply:
We have a business requirement to perform an action delete/enable/disable for all collections in a click.
On the server side, we have a transaction(DB stored procedure) performing that operation. If we don't do that in bulk we get conflicts and high-performance impact and not acceptable latency.
 Thus the API must be a bulk operation.